File: fxgl-samples/src/main/java/sandbox/view/BasicEntityViewApp.java
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package sandbox.view;

import com.almasb.fxgl.app.GameApplication;
import com.almasb.fxgl.app.GameSettings;
import com.almasb.sslogger.Logger;
import com.almasb.sslogger.LoggerLevel;
import com.almasb.sslogger.LoggerOutput;
import dev.DeveloperWASDControl;
import javafx.scene.input.KeyCode;
import javafx.scene.paint.Color;
import javafx.scene.shape.Rectangle;

import static com.almasb.fxgl.dsl.FXGL.*;

/**
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
public class BasicEntityViewApp extends GameApplication {
    @Override
    protected void initSettings(GameSettings settings) {
    }

    @Override
    protected void initInput() {
        onKeyDown(KeyCode.F, "test", () -> {
            debug("Game time: " + getGameTimer().getNow());
        });
    }

    @Override
    protected void initGame() {


        //getGameScene().addUINode(new Rectangle(getAppWidth(), getAppHeight()));

        entityBuilder()
                .at(100, 100)
                .view(new Rectangle(40, 40, Color.BLUE))
                .with(new DeveloperWASDControl())
                .buildAndAttach();
    }

    public static void main(String[] args) {
        launch(args);
    }
}

File: fxgl-samples/src/main/kotlin/sandbox/view/BasicEntityViewApp.kt
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */
package sandbox.view

import com.almasb.fxgl.app.GameApplication
import com.almasb.fxgl.app.GameSettings
import com.almasb.fxgl.dsl.FXGL.Companion.debug
import com.almasb.fxgl.dsl.FXGL.Companion.entityBuilder
import com.almasb.fxgl.dsl.FXGL.Companion.getGameTimer
import com.almasb.fxgl.dsl.FXGL.Companion.onKeyDown
import com.almasb.fxgl.dsl.components.view.GenericBarViewComponent
import dev.DeveloperWASDControl
import javafx.scene.input.KeyCode
import javafx.scene.paint.Color
import javafx.scene.shape.Rectangle

/**
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
class BasicEntityViewApp : GameApplication() {
    override fun initSettings(settings: GameSettings) {}
    override fun initInput() {
        onKeyDown(KeyCode.F, "test", Runnable { debug("Game time: " + getGameTimer().now) })
    }

    override fun initGame() {
        entityBuilder()
                .at(100.0, 100.0)
                .view(Rectangle(100.0, 100.0, Color.BLUE))
                .with(DeveloperWASDControl())
                .with(GenericBarViewComponent(0.0, 100.0, Color.GREEN, 2999.0, 4003.0))
                .buildAndAttach()
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            launch(BasicEntityViewApp::class.java, args)
        }
    }
}